app/src/common/shared/com/igalia/wolvic/utils/EnvironmentUtils.java

```diff
@@ -84,28 +86,26 @@ public static String getBuiltinEnvPath(@NonNull String envId) {
     }
 
     /**
-     * Check wether or not an external environment is ready to be used. Checks is the ouput directory exists
-     * and if it contains 6 items. We make an assumption that those items are the right images and that they
-     * follow the naming convention.
+     * Check whether or not an external environment is ready to be used. Checks if the directory exists
+     * and if it contains at least 6 supported files. We make an assumption that those files are the
      * right images and that they follow the naming convention.
      * @param context An activity context.
      * @param envId The environment id. This maps to the Remote properties JSON "value" environment property.
      * @return true is the environment is ready, false otherwise
      */
     public static boolean isExternalEnvReady(@NonNull Context context, @NonNull String envId) {
-        boolean isEnvReady = false;
         String envOutputPath = getExternalEnvPath(context, envId);
         if (envOutputPath != null) {
-            File file = new File(envOutputPath);
-            if (file.exists() && file.isDirectory()) {
-                File[] files = file.listFiles();
-                if (files != null && files.length == 6) {
-                    isEnvReady = true;
-                }
+            File envDirectory = new File(envOutputPath);
+            if (envDirectory.exists() && envDirectory.isDirectory()) {
+                File[] envFiles = envDirectory.listFiles(file -> {
+                    String fileName = file.getName().toLowerCase();
+                    return file.isFile() && Arrays.stream(SUPPORTED_ENV_EXTENSIONS).anyMatch(fileName::endsWith);
+                });
+                return envFiles != null && envFiles.length >= 6;
```

Comment:
This is indeed safer, but not strictly needed for the fix right? I was wondering it this could became a separate commit in this same PR. If they eventually fix the issue in Meta we could leave this as is and just revert the other part of the PR.